// Copyright 2018 The Swarming Authors. All rights reserved.
// Use of this source code is governed by the Apache v2.0 license that can be
// found in the LICENSE file.

syntax = "proto3";

package buildbucket.v2;

import "google/protobuf/timestamp.proto";
import "common.proto";

// A build step.
message Step {

  // A named log of a step.
  message Log {
    // Log name, standard ("stdout", "stderr") or custom ("json.output").
    // Unique within the step.
    string name = 1;

    // LogDog stream name.
    // Combined with Build.infra.logdog, can be used to load log contents.
    string logdog_stream_name = 2;

    // URL of a human-oriented page that displays log contents.
    string url = 3;
  }

  // Name of the step, unique within the build.
  // Identifies the step.
  string name = 1;

  // The timestamp when the step started.
  google.protobuf.Timestamp start_time = 2;

  // The timestamp when the step ended.
  google.protobuf.Timestamp end_time = 3;

  // Current status of the step.
  // Must be specified, i.e. not STATUS_UNSPECIFIED.
  Status status = 4;

  // Logs produced by the step.
  // Log order is up to the step.
  repeated Log logs = 5;

  // Human-oriented summary of the step provided by the step itself,
  // in Markdown format.
  //
  // V1 equivalent: combines and supersedes Buildbot's step_text and step links and also supports
  // other formatted text.
  string summary_markdown = 7;
}
